staker/block_validator.go

```diff
@@ -805,19 +808,70 @@ func (v *BlockValidator) advanceValidations(ctx context.Context) (*arbutil.Messa
 	v.reorgMutex.RLock()
 	defer v.reorgMutex.RUnlock()
 
+	for {
+		if ctx.Err() != nil {
+			return nil, ctx.Err()
+		}
+		pos := v.validated()
```

Comment:
we do want the unlock/lock above. It prevents us from capturing the lock for too long.

Reply:
Done.